src/tr/merlinscans/src/eu/kanade/tachiyomi/extension/tr/merlinscans/MerlinScans.kt

```diff
@@ -1,12 +1,431 @@
 
-import eu.kanade.tachiyomi.multisrc.mangathemesia.MangaThemesia
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
 import java.text.SimpleDateFormat
 import java.util.Locale
 
-class MerlinScans : MangaThemesia(
-    "Merlin Scans",
-    "https://merlinscans.com",
-    "tr",
-    dateFormat = SimpleDateFormat("MMMM dd, yyyy", Locale("tr")),
-)
+class MerlinScans : ParsedHttpSource() {
+
+    override val name = "MerlinScans"
+    override val baseUrl = "https://merlinscans.com"
+    override val lang = "tr"
+    override val supportsLatest = true
+
+    private val json = Json { ignoreUnknownKeys = true }
+    private val dateFormat = SimpleDateFormat("MMMM dd, yyyy", Locale("tr"))
+    private val alternativeDateFormat = SimpleDateFormat("dd.MM.yyyy", Locale.getDefault())
+
+    // Extension fonksiyonları eklendi
+    private inline fun <reified T> Response.parseAs(): T {
+        return json.decodeFromString(body.string())
+    }
+
+    private fun SimpleDateFormat.tryParse(dateStr: String): Long? {
+        return try {
+            parse(dateStr)?.time
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    private fun normalizeUrl(url: String): String {
+        return when {
+            url.contains("/series.php?slug=") -> {
+                val httpUrl = url.toHttpUrl()
+                val slug = httpUrl.queryParameter("slug") ?: return url
+                httpUrl.newBuilder()
+                    .encodedPath("/series/$slug")
+                    .query(null)
+                    .build()
+                    .encodedPath
+            }
+            url.startsWith("/series/") -> url
+            else -> url
+        }
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = "$baseUrl/all-series.php".toHttpUrl().newBuilder()
+            .addQueryParameter("sort", "views")
+            .addQueryParameter("page", page.toString())
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun popularMangaSelector() = "div.series-grid a.series-card"
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            val originalUrl = element.attr("href")
+            setUrlWithoutDomain(normalizeUrl(originalUrl))
+            title = element.select("div.series-title").text()
```

Comment:
```suggestion
            title = element.selectFirst("div.series-title")!!.text()
```

Comment applies elsewhere too.